### PR TITLE
Documentation/api

### PR DESCRIPTION
**Pull request by mrkno**
Improve API documentation as part of #192

### CODE REVIEW

File: doc/ModuleCreation.md [hunk 1]
```diff
@@ -1,198 +1,40 @@
...
+### Kassy Module Format
+For a "Hello World" example see [here](https://github.com/concierge/HelloConcierge).
+
+The basic module for a Concierge module will created as a new directory within the `modules` directory of Concierge (*not to be confused with `node_modules`</b>*).
+It will contain the following files:
+- `someModule.js` - an ES6, strict mode JavaScript file that will be the basis of the module. [See here](./api/Module.md) for the methods that this should implement.
+- `kassy.json` - a file describing your module, that can be used by Concierge to start it. [See here](./api/Kassy.json.md) for the properties that should be in this file.
+
+#### Dependencies
+Concierge has been created using [Node.JS](https://nodejs.org/). As with any Node.JS application, it is possible to depend on other Node.JS modules using `require`. Within Concierge, `require` has been extended to automatically install any dependency it does not have/cannot find from [NPM](https://www.npmjs.com/).
```
Comment (ParasDPain) [inline doc/ModuleCreation.md]: Ambiguious sentence 

> any dependency it does not have/cannot find from NPM

Implies modules that were not found on NPM.

Can be replaced with 

> Within Concierge, `require` has been extended to automatically install modules from NPM if they are not found locally.

File: doc/api/Module.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+When the module resources should be unloaded and timers should be cancelled - this is usually so your module can be updated or unloaded.
+
+**Optional**  
+**Kind**: API method  
+
+<a name="config"></a>
+## config ⇒ <code>Object</code>
+**Kind**: Object property
+
+Object for storing configuration data of the module within. Any property set on this object will automatically be persisted between restarts of your module.
```
Comment (ParasDPain) [inline doc/api/Module.md]: Perhaps we can elaborate further that this config is different from the global `config.json`

**Comment by mrkno** [comment]
@ParasDPain Sure, I can adjust those

**Comment by ParasDPain** [comment]
@mrkno Thanks! These are very minor points but thought I should mention them...should've just fixed those in my PR though.

**Comment by mrkno** [comment]
@ParasDPain should be fixed now

**Pull request merged**